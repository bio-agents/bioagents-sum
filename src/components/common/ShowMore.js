import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import * as R from 'ramda'
import FontAwesome from 'react-fontawesome'
import OverlayAgenttip from './OverlayAgenttip'

export default class ReadMore extends PureComponent {
  constructor (props) {
    super(props)

    this.state = {
      expanded: false,
    }
  }

  toggleLines = (e) => {
    // Stops <a href=# </a> from scrolling to the top of the page
    e.preventDefault()

    this.setState({
      expanded: !this.state.expanded,
    })
  }

  render () {
    const { list, more, less, lines, searchTermName, ulClassName } = this.props
    const { expanded } = this.state

    const showList = expanded ? list : R.take(lines, list)

    return (
      <div>
        <ul className={ulClassName}>
          {showList.map((item, index) => {
            return <li key={index}>
              <a href={`https://bio.agents/t?${searchTermName}=${item.term}`} target='_blank'>
                {item.term}
              </a>
              <OverlayAgenttip id='agenttip-windows' agenttipText={`EDAM: ${item.term}`}>
                <a href={item.uri} target='_blank'>
                  <FontAwesome className='icons' name='question-circle' />
                </a>
              </OverlayAgenttip>
            </li>
          })}
        </ul>
        {list.length > lines &&
          <span className='table-showmore-button'>
            <a href='#' onClick={this.toggleLines}>{expanded ? less : more}</a>
          </span>
        }
      </div>
    )
  }
}

ReadMore.defaultProps = {
  lines: 3,
  more: 'Show more',
  less: 'Show less',
}

ReadMore.propTypes = {
  list: PropTypes.array,
  more: PropTypes.string,
  less: PropTypes.string,
  lines: PropTypes.number,
}
