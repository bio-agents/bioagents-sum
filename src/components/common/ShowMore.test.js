import React from 'react'
import renderer from 'react-test-renderer'
import ShowMore from './ShowMore'

it('Renders ShowMore correctly', () => {
  const list = [
    {
      term: 'term1',
      uri: 'https://www.google.com/',
    },
    {
      term: 'term2',
      uri: 'https://www.google.com/',
    },
    {
      term: 'term3',
      uri: 'https://www.google.com/',
    },
    {
      term: 'term4',
      uri: 'https://www.google.com/',
    },
  ]

  const tree = renderer
    .create(
      <ShowMore lines={3} searchTermName='operation' list={list} ulClassName='table-list-item' />
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
