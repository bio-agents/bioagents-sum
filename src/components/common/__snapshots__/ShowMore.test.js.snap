exports[`test Renders ShowMore correctly 1`] = `
<div>
  <ul
    className="table-list-item">
    <li>
      <a
        href="https://bio.agents/?function=term1"
        target="_blank">
        term1
      </a>
      <a
        href="https://www.google.com/"
        onBlur={[Function]}
        onClick={null}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        target="_blank">
        <span
          aria-hidden={true}
          className="fa fa-question-circle icons" />
      </a>
    </li>
    <li>
      <a
        href="https://bio.agents/?function=term2"
        target="_blank">
        term2
      </a>
      <a
        href="https://www.google.com/"
        onBlur={[Function]}
        onClick={null}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        target="_blank">
        <span
          aria-hidden={true}
          className="fa fa-question-circle icons" />
      </a>
    </li>
    <li>
      <a
        href="https://bio.agents/?function=term3"
        target="_blank">
        term3
      </a>
      <a
        href="https://www.google.com/"
        onBlur={[Function]}
        onClick={null}
        onFocus={[Function]}
        onMouseOut={[Function]}
        onMouseOver={[Function]}
        target="_blank">
        <span
          aria-hidden={true}
          className="fa fa-question-circle icons" />
      </a>
    </li>
  </ul>
  <span
    className="table-showmore-button">
    <a
      href="#"
      onClick={[Function]}>
      Show more
    </a>
  </span>
</div>
`;
